pkg/engine/context/context.go

```diff
@@ -154,8 +154,11 @@ func (ctx *Context) AddUserInfo(userRequestInfo kyverno.RequestInfo) error {
 
 //AddImageDetails checks if kind is pod or pod controller, then loads details about the image
 func (ctx *Context) AddImageDetails(kindInterface interface{}, specInterface interface{}) error {
-	images := make(map[string]string)
-	imgs := make(map[string]imgInfo)
+	containerImages := make(map[string]string)
+	containerImgs := make(map[string]imgInfo)
```

Comment:
@JimBugwadia check this approach out. Here I was using a map to store the imgInfo and the key (string) was the name of the image.